### DISCUSSION TITLE
The dynamic route parameter, "params" object not supports Unicode languages other than English in In NextJS 13 (beta).

### DISCUSSION THREAD

**Discussion by RajendraAVerma** [vercel/next.js#45848] [Help] [answered]
Official Documentation of NextJS 13 (beta) describe about dynamic routes parameters in [this link](https://beta.nextjs.org/docs/api-reference/file-conventions/page#params-optional).

**Explanation :**

`app/shop/[slug]/page.js`
to access slug we use params.slug.

**Example :**

`app/shop/this-is-shop/page.js`
here `params.slug` return -> `this-is-shop`

**if we use another language other than english, for example in hindi**

`app/shop/यह-दुकान-है/page.js`
here `params.slug` return ->` %E0%A4%B9-%E0%A4%A6%E0%A5%81%E0%A4%95%E0%A4%BE%E0%A4%A8-%E0%A4%B9%E0%A5%88`.

but it should also return -> `यह-दुकान-है`.

**that's the issue. How to fix this ?**

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Do you have examples of how other frameworks handle these? I think they are called IRI, and they extend URI: https://en.wikipedia.org/wiki/Internationalized_Resource_Identifier

In Chrome, if I go to [https://en.wikipedia.org/wiki/Möbius_strip](https://en.wikipedia.org/wiki/M%C3%B6bius_strip) and run `window.location.href` on the Browser console I see: `https://en.wikipedia.org/wiki/M%C3%B6bius_strip`

In fact to be able to render `ö` I had to do it like `[https://en.wikipedia.org/wiki/Möbius_strip](https://en.wikipedia.org/wiki/M%C3%B6bius_strip)`.

From another perspective, if you manually paste `https://en.wikipedia.org/wiki/Möbius_strip` into the browser URL it'll still show `ö`, but if you copy the same string you just input into the browser URL, you'll get `https://en.wikipedia.org/wiki/M%C3%B6bius_strip` in your clipboard.

**Reply by RajendraAVerma** [reply]
hey @icyJoseph thanks for your response.

I know but thing is work on `pages` dir on nextJS project.
_for example :_  
in `pages/post/[slug].js`
```
export async function getServerSideProps({ params }) {
   const slug = params.slug     // <---- HERE, it return exact text, not IRI extended.
}
```

but question is if it work on this side, why not on other side.

Do you have any Idea ?
or Any way to convert URI to actual text ?

**Reply by icyJoseph** [reply]
I see, I could think of using `decodeURI` or `decodeURIComponent`

![Screenshot 2023-02-13 at 14 26 40](https://user-images.githubusercontent.com/21013447/218470271-3b98fd9f-29da-4abc-a23e-5fbdcb1aad3d.png)

Not sure why there's divergence between app and pages, but perhaps some library is not used anymore, or maybe since they want developers to have more control over their server, they've loosed up some of the assertions or conversions done by their APIs.

**Reply by RajendraAVerma** [reply]
Thanks @icyJoseph
we can decode IRI to original text through `decodeURI()`,
honestly you saved my day ❤️
